webhooks/validating/configuration/src/main.rs

```diff
@@ -521,6 +401,113 @@ mod tests {
     }
     "#;
 
+    const VALID_BROKER_POD_SPEC: &str = r#"
+    "brokerPodSpec": {
+        "containers": [
+            {
+                "image": "image",
+                "name": "name",
+                "resources": {
+                    "limits": {
+                        "{{PLACEHOLDER}}": "1"
+                    }
+                }
+            }
+        ],
+        "imagePullSecrets": [
+            {
+                "name": "name"
+            }
+        ]
+    }
+    "#;
+
+    // Valid JSON but invalid akri.sh/v0/Configuration when inserted into
+    // brokerSpec of ADMISSION_REVIEW OR EXTENDED_ADMISSION_REVIEW constants.
+    // Misplaced `resources`
+    //   Valid: .request.object.spec.brokerSpec.brokerPodSpec.containers[*].resources
+    // Invalid: .request.object.spec.brokerSpec.brokerPodSpec.resources
+    const INVALID_BROKER_POD_SPEC: &str = r#"
```

Comment:
i think it woud be good to have an invalid test where both job and pod specs are defined.

Reply:
I had the wrong thinking around this. Currently, our CRD does not specify `oneOf` with regards to `brokerSpec`. It seems that we cannot use the  [`oneOf`](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema) structural schema specifier since we need to set `x-kubernetes-preserve-unknown-fields: true`. We should do more investigation here to see if we can limit in our Configuration CRD specifying both types of brokers -- that way the error isnt caught by our application and they error out. For now, here is the test that asserts that serde will catch this and error out the Agent: 20422dc